src/utils/components/UserCredentials/useSSHCommand.ts

```diff
@@ -0,0 +1,46 @@
+import { InfrastructureModel, modelToGroupVersionKind } from '@kubevirt-ui/kubevirt-api/console';
+import { IoK8sApiCoreV1Service } from '@kubevirt-ui/kubevirt-api/kubernetes';
+import { V1VirtualMachineInstance } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { getCloudInitCredentials } from '@kubevirt-utils/resources/vmi';
+import { K8sResourceCommon, useK8sWatchResource } from '@openshift-console/dynamic-plugin-sdk';
+
+export type useSSHCommandResult = {
+  command: string;
+  user: string;
+  sshServiceRunning: boolean;
+};
+
+const getInfrastructureAPIURL = (infrastructure: any): string =>
+  infrastructure && infrastructure?.status ? infrastructure?.status.apiServerURL : undefined;
```

Comment:
I'm guessing, infrastructure?.status is enough

Reply:
and also infrastructure?.status?.apiServerURL

Reply:
hmm, maybe even just return: infrastructure?.status.apiServerURL, as u return undefined if its don't exist.. and if  infrastructure?.status?.apiServerURL don't exist, it already undefined.